src/windows/win-dumpconfigurator.ps1

```diff
@@ -0,0 +1,70 @@
+ . .\src\windows\common\setup\init.ps1
+<#
+.SYNOPSIS
+    This script will change the dump configuration without requiring a reboot.
+
+    Prerequisites (Links below):     
+    Kdbgctrl.exe (From the Debugging Tools for Windows.)
+ 
+ 
+.NOTES
+    Name: win-dumpconfigurator.ps1
+    Author: Microsoft CSS
+    Version: 1.1
+    Created: 2021-Mar-1
+ 
+ 
+.EXAMPLE
+    ./win-dumpconfigurator.ps1 -DumpType full -DumpFile C:\Dumps\Memory.dmp -DedicatedDumpFile D:\dd.sys
+ 
+ 
+.LINK
+    Debugging Tools for Windows -- https://docs.microsoft.com/en-us/windows-insider/flight-hub/
+#>
+Param(
+[parameter()] [switch]$OneDump,
+[parameter()] [ValidateSet("active", "automatic", "full", "kernel", "mini" )] [string]$DumpType,
+[parameter()] [string]$DumpFile,
+[parameter()] [string]$DedicatedDumpFile="C:\DedicatedDumpFile.sys",
+[parameter()] [string]$VMName,
+[parameter()] [string]$ResourceGroup)
+
+#For testing purposes
+#$DumpType = 'full'
+#$DedicatedDumpFile="C:\DedicatedDumpFile.sys"
+
+if (!$DumpType) 
+{
+    Write-Host "DumpType, VMname, and Resource group *MUST* all be specified. "
+    Write-Output "Valid dump types include active, automatic, full, kernel, or mini."
+    break
+}
+
+## Add any registry prereqs to the script
+## For example, if you want to use a DedicatedDumpFile, you'll need to make sure the key exists.
+$CrashCtrlPath = "HKLM:\SYSTEM\CurrentControlSet\Control\CrashControl"
```

Comment:
Does the symbolic link for CurrentControlSet work when the VM is not running? Or are there scenarios where CurrentControlSet might not work?

Reply:
This script is intended to be run when the VM is up and running. I have not added any functionality for an offline mode, yet. Would you recommend adding that to this script, or should we have a different approach of changing any registry value via an offline mode. Maybe for example, have a OfflineReg.ps1 script where you could pass arguments to indicate what you want to do? Delete this key, update that key, etc.

Reply:
If the script is meant to be run when VM is up and running then it should be fine.